assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,163 @@
+    this.lastSavedText = this.initialText
+  }
+
+  this.close = () => {
+    self.editTextError.hide()
+  }
+
+  this.areChangesSaved = () => {
+    return this.editText.val().trim() === this.lastSavedText.trim()
+  }
+
+  this.save = (languageSelected) => {
+    const newText = this.editText.val().trim()
+    if (this.areChangesSaved()) {
+      this.close()
+      return
+    }
+
+    if (!this.hasDefaultText && !(languageSelected === '' || languageSelected === 'default')) {
+      return
+    }
+
+    if (languageSelected === '' || languageSelected === 'default') {
+      let url
+      if (this.programSection === 'name') {
+        url = this.pathEditName
+      } else if (this.programSection === 'description') {
+        url = this.pathEditDescription
+      } else if (this.programSection === 'credits') {
+        url = this.pathEditCredits
+      }
+
+      return new Promise(function (resolve, reject) {
+        $.ajax({
+          url: url,
+          type: 'put',
+          data: { value: newText },
+          success: function (data) {
+            const statusCode = parseInt(data.statusCode)
+            if (statusCode === 527 || statusCode === 707) {
+              showError(data)
+              reject(statusCode)
+            } else {
+              resolve(statusCode)
+            }
+          },
+          error: function (error) {
+            if (error.status === 422) {
+              error.message = error.responseText
+              showError(error)
+              reject(error.status)
+            } else if (error.status === 401) {
+              reject(error.status)
+            }
+          }
+        })
+      })
+    } else if (newText === '') {
+      return this.customTranslationApi.deleteCustomTranslation(
+        this.programId,
+        languageSelected,
+        deleteCustomTranslationSuccess,
+        showError
+      )
+    } else {
+      return this.customTranslationApi.saveCustomTranslation(
+        this.programId,
+        newText,
+        languageSelected,
+        saveCustomTranslationSuccess,
+        showError
+      )
+    }
+  }
+
+  function showError (error) {
+    self.editText.addClass('danger')
+    self.editTextError.show()
+    self.editTextError.text(error.message)
+  }
+
+  function deleteCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationDeleted, language)
+    self.close()
+  }
+
+  function saveCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationSaved, language)
+    self.close()
+  }
+
+  function getCustomTranslationSuccess (data) {
+    self.textLoadingSpinner.hide()
+    self.editText.val(data)
+    self.lastSavedText = data
+    self.editText.removeAttr('disabled')
+  }
+
+  function getCustomTranslationError () {
+    self.textLoadingSpinner.hide()
+    self.editText.val('')
+    self.lastSavedText = ''
+    self.editText.removeAttr('disabled')
+  }
+
+  this.getNewText = (language) => {
+    if (language === 'default') {
```

Comment:
LOL this is a bug fix? if so, is there a new test for this? I didn't check

Reply:
This isn't a bug fix. Before if there wasn't anything defined for the text field, then the language selector didn't show. However, we need to show it now because they could have something defined for description and not credits. So, now the text field is disabled if there is not a default text defined and there is a different language selected if that makes sense. For example, if a default description is defined, but no default credits. When the language selected is french, then the description will be enabled and the credit will be disabled

Reply:
IC. Do we really need that complexity of enabling certain fields and not others? I think it's fine to just let the user define whatever they want.

Also i think there was a bug. 
![image](https://user-images.githubusercontent.com/37479705/154365956-eb19a6da-bd9b-4353-a7e0-84cdfed300d9.png)
That's what happens if you try to edit the title, change to a language then change back to default.

Reply:
Previously, you couldn't define a translation without defining a default. So, I was just trying to continue the previous logic by disabling the fields that don't have a default. I don't really think it makes sense to be able to define a translation without a default as well.

Also, I just ran that test case and I didn't get that API call. What did you click when the keep or discard text prompt came up?

Reply:
when i did it, i didn't edit anything, i just went to the edit page and directly changed the language, then change back to default

Reply:
Oh, okay. I just ran that test as well and I still didn't get that API call. Is that on your local or on prod?

Reply:
Bug should be fixed in this PR